module/solutions/helper.js

```diff
@@ -261,79 +255,100 @@ module.exports = class SolutionsHelper {
             message : constants.apiResponses.SOLUTION_NOT_FOUND
           });
         }
-
+       
         if( programData[0].scope ) {
           
           let currentSolutionScope = JSON.parse(JSON.stringify(programData[0].scope));
-
+          
           if( Object.keys(scopeData).length > 0 ) {
-
             if( scopeData.entityType ) {
-              
-              let entityType =  await entityTypesHelper.entityTypesDocument(
-                {
-                  name : scopeData.entityType
-                },
-                ["name","_id"]
-              );
-          
-              currentSolutionScope.entityType = entityType[0].name;
-              currentSolutionScope.entityTypeId = entityType[0]._id;
-
+              let bodyData = { "type" : scopeData.entityType }
```

Comment:
@VISHNUDAS-tunerlabs please check if we can avoid this blank search.

Reply:
@aks30 checking if the entity type got from request is valid or not